### DISCUSSION TITLE
portainer oauth issue - cannot fetch token: 403 Forbidden

### DISCUSSION THREAD

**Discussion by lr7skquo** [authelia/authelia#5715] [Troubleshooting] [answered]
I've configured authelia oidc for portainer, everything seems ok in authelia logs, however I'm getting an "Unauthorized" in portainer UI.
I've checked the logs in portainer and I found "oauth2: cannot fetch token: 403 Forbidden".
[authelia_logs.txt](https://github.com/authelia/authelia/files/12189867/authelia_logs.txt)
[portainer_logs.txt](https://github.com/authelia/authelia/files/12189868/portainer_logs.txt)

both authelia and portainer are behand the NPM. The NPM settings as below:
![image](https://github.com/authelia/authelia/assets/4255110/7c8e3443-08b4-4a61-86ce-e97dc667f58b)
![image](https://github.com/authelia/authelia/assets/4255110/64876343-d351-4215-9821-b6de165cbaeb)

authelia config as below:
[configuration - Copy.yml.txt](https://github.com/authelia/authelia/files/12189912/configuration.-.Copy.yml.txt)
[users_database.yml.txt](https://github.com/authelia/authelia/files/12189913/users_database.yml.txt)

portainer settings:
![image](https://github.com/authelia/authelia/assets/4255110/102726f2-48f0-4afa-9862-78002eaa07c3)
![image](https://github.com/authelia/authelia/assets/4255110/14d46a1d-4810-416c-94a4-a10df13a4b62)
![image](https://github.com/authelia/authelia/assets/4255110/a1073920-52ab-4117-8a85-a8fcfe4d3fa3)

How can I further debug this?
Thank you

**Comment by james-d-elliott** [comment, marked as answer]
Looks like potentially portainer can't communicate with authelia, there are no access request logs (indicative that the request to access the token never occurs). Maybe a DNS issue?

**Reply by lr7skquo** [reply]
Hi James,

You are right -- portainer can't communicate with authelia.
There is a misconfigured WAF rule in Cloudflare that blocks portainer from connecting the sub domain of authelia. After correct that rule, everything is fine.

Thank you very much.

**Reply by james-d-elliott** [reply]
No worries, glad I pointed you in the right direction.
